docs: drop `newEvent()` compatibility function

PowerKiKi opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Users are confused by the use of `newEvent` that looks like a typo, and whose origin is unclear.

Issue Number: #23370


## What is the new behavior?

Use the straightforward `new Event()` syntax.

Because PhantomJS has been deprecated since March 2018, and `newEvent`
is very confusing for newcomers that read the testing documentation,
we remove it entirely, and instead assume most, if not all, newcomers
will run tests in Chrome as it is the default.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

gkalpak commented on aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:
```diff
@@ -275,7 +275,7 @@ describe('demo (with TestBed):', () => {
         // dispatch a DOM event so that Angular learns of input value change.
         // then wait while ngModel pushes input.box value to comp.name
-        input.dispatchEvent(newEvent('input'));
+        input.dispatchEvent(new Event('input'));
```
Does this work on all supported browsers (including IE 9-11)?
If not we should either keep (and document) the helper or mention the limitation and the workaround for older browsers.

PowerKiKi replied:
No, it does not work in any IE versions [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event#Browser_compatibility).

However since all the code updated here is strictly a _sample_ for _testing_, and thus will never reach production, I assumed it was OK to drop support for those browser. Edge has now been available for 4 years. And I doubt many people actually run tests on IE out there. And for the few people who actually do I am sure they are very well aware of all the limitations and edge cases that comes with IE.

I don't think (this section of) the documentation is the right place to address issues for legacy browsers that concern a very small minority of users. Especially since this is only testing code, not production code.

gkalpak replied:
I don't necessarily agree (personally I would expect the docs examples to work on all supported browsers), but I'll leave to the docs team to decide.

pkozlowski-opensource replied:
I would side with @gkalpak - I don't think that we are gaining much by removing this `newEvent` helper. We can do this as we stop supporting browsers that need it.

PowerKiKi replied:
By removing the helper, you would gain in clarity and reduce the noise in the issue tracker, see #23370, #36967 and #32001.

But feel free to close this PR if it does not align with the project goals.

daicoden replied:
It's very weird to have documentation and a comment `//not provided by angular`, with no pointers to what/why/where it comes from, followed the rabbit hole and ended up here.

2 cents from a random user who couldn't figure out what to do from the docs - examples in documentation should work, and the ones from the testing document don't currently.

If you're saying that angular supports I.E 9-11, then it would be great if the angular testing framework provided support for I.E 9-11, and the docs use whatever the first class API is.

It's a small function, so another solution would be to put a note in the doc with code to copy into your project but right now it's confusing when reading the examples.

gkalpak replied:
I generally agree. That's why in my original comment (https://github.com/angular/angular/pull/37251#discussion_r429110025) I said _"we should either keep **(and document)** the helper or **mention the limitation and the workaround** for older browsers"_.

PowerKiKi replied:
do you mean add a single line comment above each `new Event` call ? something like:

```
// For compatibility with older browsers, such a IE11, see https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill
input.dispatchEvent(new Event('input'));
```

gkalpak replied:
Yeah, something like that. I am not sure if it makes sense to mention it in the guide in addition/instead of as a comment in the source code. (Something for the docs team to decide.)

PowerKiKi replied:
Then I'll wait for the official decision from the doc team, before changing anything

aikidave replied:
Hi @PowerKiKi! Thanks for this PR! This does seem to be a tricky issue. On the one hand, we want to keep code as simple as possible. On the other hand, we don't want to confuse anyone.

I like your idea of adding a single line comment above each use of `new Event().` It's a bit of extra work to add it, but I think it will help add greater clarity for other developers.

Happy to discuss further if you want!

PowerKiKi replied:
I added a single line comment everywhere as discussed

gkalpak reviewed (dismissed):
Thx, @PowerKiKi! I've left a comment (i.e. I don't think we are ready to just drop the helper).

gkalpak commented on aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:
```diff
@@ -273,9 +273,10 @@ describe('demo (with TestBed):', () => {
+        // Dispatch a DOM event so that Angular learns of input value change.
         // then wait while ngModel pushes input.box value to comp.name
-        input.dispatchEvent(newEvent('input'));
+        // For compatibility with older browsers, such a IE11, see https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill
```
I find this a little confusing. We send users to a page that shows them how to polyfill `CustomEvent`, but in the code below we are not using `CustomEvent`. Mayse say something like:

> In older browsers, such as IE, you might need to a `CustomEvent` instead.
> See https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill.

PowerKiKi replied:
Updated accordingly

gkalpak commented:
@PowerKiki, can you please rebase this on master and squash all commits into one 🙏

mary-poppins commented:
You can preview b2ecd3a at https://pr37251-b2ecd3a.ngbuilds.io/.
You can preview d063ad7 at https://pr37251-d063ad7.ngbuilds.io/.
You can preview 58b991e at https://pr37251-58b991e.ngbuilds.io/.

gkalpak commented on aio/content/examples/testing/src/app/demo/demo.testbed.spec.ts:
```diff
@@ -273,9 +273,10 @@ describe('demo (with TestBed):', () => {
+        // Dispatch a DOM event so that Angular learns of input value change.
         // then wait while ngModel pushes input.box value to comp.name
-        input.dispatchEvent(newEvent('input'));
+        // In older browsers, such as IE, you might need a CustomEvent instead, see https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill
```
This long line will not look good in the code-snippets: 

![new-event-wrapped](https://user-images.githubusercontent.com/8604205/83509322-eed1c100-a4d3-11ea-9d77-c9622cdee328.png)

Please change it to something like (here and in the other files):

```suggestion
        // In older browsers, such as IE, you might need a CustomEvent instead. See
        // https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent#Polyfill
```

PowerKiKi replied:
done

gkalpak reviewed (changes requested):
@PowerKiki, can you please rebase this on master and squash all commits into one (that should make CI happy) 🙏
Also, please change the commit message to `docs: ...` (instead of `test(docs-infra): ...`).

PowerKiKi commented:
@gkalpak, I rebased and squashed, feel free to review again.

mary-poppins commented:
You can preview e6c4424 at https://pr37251-e6c4424.ngbuilds.io/.

aikithoughts reviewed (approved):
Changes look good! Thank you, @PowerKiKi for your perseverance! 

Reviewed-for: global-docs-approvers

kapunahelewong commented:
@PowerKiKi, could CircleCI be failing because you've edited the spec files?

PowerKiKi commented:
I couldn't tell, because I cannot access CircleCI logs. And I am not willing to give write access to all my unrelated public repos to see logs.

PowerKiKi committed:
docs: drop `newEvent()` compatibility function

Because PhantomJS has been deprecated since March 2018, and `newEvent`
is very confusing for newcomers that read the testing documentation,
we remove it entirely, and instead assume most, if not all, newcomers
will run tests in Chrome as it is the default.

Fixes #23370

mary-poppins commented:
You can preview 7db4a92 at https://pr37251-7db4a92.ngbuilds.io/.

gkalpak reviewed (approved):
LGTM :+1: Thx, @PowerKiKi!

FYI: I rebased it on latest master to make CI happy.

gkalpak commented:
Somehow this slipped through the cracks (sorry for that, @PowerKiKi).
I rebased on master, resolved conflicts and marked it for merging (as soon as CI is happy) 🚀

mary-poppins commented:
You can preview 11e6fdc at https://pr37251-11e6fdc.ngbuilds.io/.
